### PR TITLE
Elastic 6.3.0 support; closes #168

### LINKED ISSUE #168: Unable to install plugin for elasticsearch 6.3.0
Could please help me in installing plugin for es 6.3.0 instead of es lower versions.

### PR DESCRIPTION
**Pull request by nomoa**
Fixes #168

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,7 @@
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.5-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4.1-bin.zip
+zipStoreBase=GRADLE_USER_HOME
+distributionSha256Sum=6ac2f8f9302f50241bf14cc5f4a3d88504ad20e61bb98c5fd048f7723b61397e
```
Comment (purbon) [inline gradle/wrapper/gradle-wrapper.properties]: sorry for the might be stupid question, what is this sha for?

File: src/main/java/com/o19s/es/ltr/feature/store/StoredLtrModel.java [hunk 2]
```diff
@@ -164,7 +165,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (modelAsString) {
             builder.value(rankingModel);
         } else {
-            try (XContentParser parser = JsonXContent.jsonXContent.createParser(EMPTY, rankingModel)) {
+            try (XContentParser parser = JsonXContent.jsonXContent.createParser(EMPTY,
+                    LoggingDeprecationHandler.INSTANCE, rankingModel)
```
Comment (purbon) [inline src/main/java/com/o19s/es/ltr/feature/store/StoredLtrModel.java]: again, sorry for the might be stupid question. Why this is logged as deprecated?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (nomoa) [reply gradle/wrapper/gradle-wrapper.properties]: I copied it from upstream, I guess it's an extra check to make sure that binary downloaded is valid (probably security concerns?)

File: src/main/java/com/o19s/es/ltr/feature/store/StoredLtrModel.java (hunk 2 above)
Comment (nomoa) [reply src/main/java/com/o19s/es/ltr/feature/store/StoredLtrModel.java]: This is the new signature for createParser, I suppose that now they force us to specify the deprecation policy. (this is part of https://github.com/elastic/elasticsearch/issues/28504 & https://github.com/elastic/elasticsearch/pull/28697)

File: src/main/java/com/o19s/es/ltr/feature/store/StoredFeature.java [hunk 3]
```diff
@@ -114,7 +114,7 @@ public StoredFeature(String name, List<String> params, String templateLanguage,
     }
 
     public StoredFeature(String name, List<String> params, String templateLanguage, XContentBuilder template) {
-        this(name, params, templateLanguage, Objects.requireNonNull(template).bytes().utf8ToString(), false);
+        this(name, params, templateLanguage, BytesReference.bytes(Objects.requireNonNull(template)).utf8ToString(), false);
```
Comment (nomoa) [inline src/main/java/com/o19s/es/ltr/feature/store/StoredFeature.java]: should be Strings.toString

**Comment by softwaredoug** [comment]
Looks good 👍 smoke tested with demo on ES 6.3.1. Merging...

**Pull request merged**